File: pymdurs/examples/cadastre_from_ign.py
"""
Example: Download Cadastre (parcel) data from IGN API using rsmdu

This example demonstrates how to:
1. Create a Cadastre instance
2. Set a bounding box
3. Download cadastre from IGN API
4. Get GeoJSON data
5. Save to GPKG file
"""

import json

import pymdurs


def main():
    print("🏛️  Loading Cadastre from IGN API...")

    # Create Cadastre instance
    cadastre = pymdurs.geometric.Cadastre(output_path="./output")

    # Set bounding box (La Rochelle area, France)
    # Format: min_x, min_y, max_x, max_y (WGS84, EPSG:4326)
    cadastre.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699)

    # Set CRS (optional, defaults to EPSG:2154)
    cadastre.set_crs(2154)

    geo = cadastre.geo_core
    print(f"📦 Bounding box set")
    print(f"🗺️  CRS: {geo.epsg}")
    print(f"📁 Output path: {geo.output_path}")

    # Run cadastre processing: downloads from IGN API and parses GeoJSON
    print("⏳ Downloading cadastre from IGN API...")
    cadastre = cadastre.run()

    # Get GeoJSON (equivalent to to_gdf() in Python)
    print("📊 Getting GeoJSON data...")
    geojson = cadastre.get_geojson()

    if geojson and "features" in geojson:
        num_features = len(geojson["features"])
        print(f"✅ Loaded {num_features} cadastre parcels")
    else:
        print("✅ Cadastre data loaded")

    # Save to GPKG
    print("💾 Saving to GPKG...")
    cadastre.to_gpkg(name="cadastre")

    print(f"✅ Cadastre processing complete!")
    print(f"📁 Output path: {cadastre.get_output_path()}")

    return cadastre


if __name__ == "__main__":
    cadastre = main()

File: rsmdu/examples/cadastre_from_ign.rs
use anyhow::Result;
use rsmdu::geometric::cadastre::Cadastre;

/// Example: Loading Cadastre (parcel) data from IGN API
/// Following Python example from pymdu.geometric.Cadastre
fn main() -> Result<()> {
    println!("=== Example: Loading Cadastre from IGN API ===\n");

    // Create Cadastre instance
    // Python: cadastre = Cadastre(output_path='./')
    let mut cadastre = Cadastre::new(Some("./output".to_string()))?;

    // Set bounding box (La Rochelle, France)
    // Python: cadastre.bbox = [-1.152704, 46.181627, -1.139893, 46.18699]
    cadastre.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699);

    println!("Bounding box set:");
    println!("  - Longitude: -1.152704 to -1.139893");
    println!("  - Latitude: 46.181627 to 46.18699");
    println!("  - Area: La Rochelle, France");
    println!("  - Format: WGS84 (EPSG:4326)\n");

    // Run cadastre processing
    // Python: cadastre = cadastre.run()
    println!("Downloading and processing Cadastre from IGN API...");
    let cadastre_result = cadastre.run()?;

    println!("\nCadastre processed successfully!");

    // Get GeoJSON (equivalent to to_gdf() in Python)
    if let Some(geojson) = cadastre_result.get_geojson() {
        match geojson {
            geojson::GeoJson::FeatureCollection(fc) => {
                println!("  - Number of parcels: {}", fc.features.len());
            }
            _ => {
                println!("  - GeoJSON loaded (non-FeatureCollection format)");
            }
        }
    }

    // Save to GeoJSON
    // Python: cadastre.to_geojson(name="cadastre")
    println!("\nSaving to GeoJSON...");
    cadastre_result.to_geojson(None)?;

    println!("\n✅ Processing complete!");
    println!("  - Output file: {:?}", cadastre_result.get_output_path());

    Ok(())
}

